engine/class_modules/warlock/sc_warlock_demonology.cpp

```diff
@@ -250,7 +244,14 @@ struct hand_of_guldan_t : public demonology_spell_t
   {
     demonology_spell_t::impact( s );
 
-    impact_spell->execute();
+    impact_spell->execute();    
```

Comment:
This probably needs to set a target too (or use the execute_on_target fn).

Reply:
Would the target not be set by the call up to `action_t::impact()` ?

https://github.com/simulationcraft/simc/blob/d6699509e1fa07a7263ec3764dfc8ed4a26a4478/engine/action/sc_action.cpp#L3925-L3942

Reply:
That's `impact_action`, not `impact_spell`

Reply:
This is the action_t::impact function, which as far as I am aware is what is called up to via 
`sc_warlock_demonology::impact()` (Only calls up)
->`sc_warlock::impact()` (Only calls up)
-> `sc_spell_t::impact()` (Not overridden)
->`sc_spell_base_t::impact()` (Not overridden) 
-> `sc_action::impact()`

Is there some interaction along the way that would prevent this from occurring?

Reply:
There's nothing in the code you linked that ensures `impact_spell::target` is the same as `s->target`.

Reply:
Perhaps I misunderstand
the `impact_spell` is of type `hog_impact_t`, which inherits from `demonology_spell_t`, which as we've shown above inherits up to `action_t`

The target variable of `impact_spell` does not appear to be overridden either, so it would be the reference set in `action_t`, which is set by the call at the bottom of this code to `impact_action`'s `::set_target()` function, is it not?

https://github.com/simulationcraft/simc/blob/d6699509e1fa07a7263ec3764dfc8ed4a26a4478/engine/action/sc_action.cpp#L4518-L4527

So if the impact itself has its target driven by the state passed to it (always in reference to the `state->target` in `action_t`), and the `impact_spell` the impact function creates has its target set to the state's target, So the value of `impact_spell->target` should be the same as `s->target` regardless of which `impact_spell` variable we are talking about, right?

I'm confused which target variable we believe needs to be set here. I just wonder because up until this point it has been functional and hitting the target without setting it.

Reply:
Should be resolved by #6243